src/essence/Tools/Isochrone/IsochroneTool_Manager.js

```diff
@@ -0,0 +1,443 @@
+           "Resolution",
+           this.sections.data
+        ).on("change", e => this.handleInput(e, "resolution", 3));
+        
+        for(const dataType of this.modelProto.requiredData) {
+            this.options[dataType + "_source"] = 0;
+            createDropdown(
+                dataType + " source",
+                this.sections.data,
+                this.dataSources[dataType].map(s => s.name)
+            ).on("change", e => {
+                this.options[dataType + "_source"] = parseInt(e.target.value);
+                this.updateResolutionRange();
+                this.handleInput(e, null, 3);
+            });
+        }
+
+        this.updateResolutionRange();
+        
+        createInputWithUnit(
+            "Max " + this.modelProto.costName,
+            this.sections.model,
+            this.modelProto.costUnitSymbol,
+            this.modelProto.defaultCost,
+            `min="0" step="1"`
+        ).on("change", e => this.handleInput(e, "maxCost", parseFloat(e.target.value) < this.options.maxCost ? 1 : 2));
+        this.options.maxCost = this.modelProto.defaultCost;
+
+        this.model.createOptions(this.sections.model, (e, action) => this.handleInput(e, null, action));
+
+        return reuseData;
+    }
+
+    focus() {
+        this.optionEls.root.addClass("focused");
+    }
+
+    unfocus() {
+        this.optionEls.root.removeClass("focused");
+    }
+    
+    updateResolutionRange() {
+        let max = GLOBAL_MAX_RESOLUTION;
+        let min = GLOBAL_MIN_RESOLUTION;
+        for(const dataType of this.modelProto.requiredData) {
+            const sourceObj = this.dataSources[dataType][this.options[dataType + "_source"]];
+            max = Math.min(max, sourceObj.maxResolution);
+            min = Math.max(min, sourceObj.minResolution);
+        }
+        const newVal = Math.max(min, Math.min(max, this.options.resolution));
+        this.optionEls.resolution.empty();
+        for(let e = min; e <= max; e++) {
+            $(`<option value=${e}${e === newVal ? " selected" : ""}>${e}</option>`)
+                .appendTo(this.optionEls.resolution);
+        }
+        this.minResolution = min;
+        this.maxResolution = max;
+        this.optionEls.resolution.value = newVal;
+        this.options.resolution = newVal;
+    }
+
+    /******************** ISOCHRONE GENERATION ********************/
+
+    setBounds() {
+        this.currentStage = 3;
+
+        //There's almost certainly a smart, not brute-force mathematical solution to this problem...
+        //TODO work out what that is
+        const startPoint = Map_.map.project(this.start, this.options.resolution);
+        const startTile = startPoint.divideBy(256);
+        let min = startTile.floor().multiplyBy(256);
+        let max = startTile.ceil().multiplyBy(256);
+
+        const isWithinBounds = (ptX, ptY) => {
+            const ptLatLng = Map_.map.unproject([ptX, ptY], this.options.resolution);
```

Comment:
explain why you can't just get the min and max from using the start tile and radius? can't you convert the tiles to latlng using resolution?

Reply:
got a more sensible constant-time implementation of this behavior now.
This was originally implemented with Leaflet's `LatLng.toBounds`, before it became clear to me that working in lat/long was inadvisable for projection support, and `LatLngBounds` (the return type of that function) in particular are universally unsafe on polar maps. The code went away, but the assumption that I had to work with lat/long for this problem stuck around a while longer.